### PR TITLE
Generate solacc reports

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Currently, `solacc` outputs advices to the console, and does not keep track of linting time.
This PR:
 - dumps advices to an 'advices.txt' file
 - collects stats and dumps them to a 'stats.json' file (in the form of a json-like file, with 1 json object per line)
 - uploads the above 

### Linked issues
None

### Functionality
None

### Tests
- [x] manually tested:

Sample solacc run:
Screenshot 2024-09-27 at 16 44 22

Sample stats.json (expanded):
```
{
  "run_id": "1",
  "name": "ab-testing",
  "start_timestamp": "2024-09-27 10:16:02.512363+00:00",
  "end_timestamp": "2024-09-27 10:17:07.622161+00:00",
  "files_count": 6,
  "files_size": 34934
}
{
  "run_id": "1",
  "name": "adverse-drug-events",
  "start_timestamp": "2024-09-27 10:17:08.669225+00:00",
  "end_timestamp": "2024-09-27 10:17:09.399495+00:00",
  "files_count": 5,
  "files_size": 48743
}
{
  "run_id": "1",
  "name": "als-recommender",
  "start_timestamp": "2024-09-27 10:17:09.565942+00:00",
  "end_timestamp": "2024-09-27 10:17:12.039422+00:00",
  "files_count": 6,
  "files_size": 62750
}
```

Sample advices.txt:
```
./dist/ab-testing/4. Real time inference.py:76:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/4. Real time inference.py:139:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/4. Real time inference.py:139:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/RUNME.py:1:1: [library-install-failed] Unsupported 'pip' command: DBTITLE
./dist/ab-testing/5. AB testing metrics.py:50:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/5. AB testing metrics.py:50:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/5. AB testing metrics.py:51:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/5. AB testing metrics.py:51:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/5. AB testing metrics.py:54:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/5. AB testing metrics.py:54:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/5. AB testing metrics.py:58:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/5. AB testing metrics.py:58:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/2. Model training.py:34:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/2. Model training.py:34:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/2. Model training.py:36:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/2. Model training.py:37:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/2. Model training.py:37:0: [rdd-in-shared-clusters] RDD APIs are not supported on UC Shared Clusters. Rewrite it using DataFrame API
./dist/ab-testing/2. Model training.py:38:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/2. Model training.py:41:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/ab-testing/2. Model training.py:41:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/ab-testing/5. AB testing metrics.py:22:0: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/5. AB testing metrics.py:36:27: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/5. AB testing metrics.py:90:2: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/5. AB testing metrics.py:97:2: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/risk_demo.dbdash:1:0: [unknown-language] Cannot detect language for /home/runner/work/ucx/ucx/dist/ab-testing/risk_demo.dbdash
./dist/ab-testing/1. Introduction.py:72:2: [direct-filesystem-access] The use of direct filesystem references is deprecated: /tmp/german_credit_data.csv
./dist/ab-testing/4. Real time inference.py:10:0: [direct-filesystem-access] The use of direct filesystem references is deprecated: /FileStore/tmp/streaming_ckpnt_risk_demo
./dist/ab-testing/4. Real time inference.py:10:14: [direct-filesystem-access] The use of direct filesystem references is deprecated: /FileStore/tmp/streaming_ckpnt_risk_demo
./dist/ab-testing/4. Real time inference.py:29:5: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/4. Real time inference.py:214:32: [direct-filesystem-access] The use of direct filesystem references is deprecated: /FileStore/tmp/streaming_ckpnt_risk_demo
./dist/ab-testing/4. Real time inference.py:236:2: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/4. Real time inference.py:254:27: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/2. Model training.py:53:22: [direct-filesystem-access] The use of direct filesystem references is deprecated: /Users/Uninferable/german_credit_experiment
./dist/ab-testing/2. Model training.py:66:5: [default-format-changed-in-dbr8] The default format changed in Databricks Runtime 8.0, from Parquet to Delta
./dist/ab-testing/2. Model training.py:197:67: [direct-filesystem-access] The use of direct filesystem references is deprecated: /tmp/pr-curve-model-a.png
./dist/ab-testing/2. Model training.py:229:67: [direct-filesystem-access] The use of direct filesystem references is deprecated: /tmp/pr-curve-model-b.png
./dist/adverse-drug-events/RUNME.py:1:1: [library-install-failed] Unsupported 'pip' command: DBTITLE
./dist/adverse-drug-events/01-ade-extraction.py:23:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/adverse-drug-events/01-ade-extraction.py:23:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/adverse-drug-events/02-ade-analysis.py:13:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters
./dist/adverse-drug-events/02-ade-analysis.py:13:0: [legacy-context-in-shared-clusters] sc is not supported on UC Shared Clusters. Rewrite it using spark
./dist/adverse-drug-events/02-ade-analysis.py:14:0: [jvm-access-in-shared-clusters] Cannot access Spark Driver JVM on UC Shared Clusters

```

### CODE REVIEW

File: tests/integration/source_code/solacc.py [hunk 1]
```diff
@@ -72,23 +77,41 @@ def _collect_uninferrable_count(advices: list[LocatedAdvice]):
```
Comment (JCZuurmond) [inline tests/integration/source_code/solacc.py:78]: Add return types

File: tests/integration/source_code/solacc.py [hunk 2]
```diff
@@ -20,6 +23,8 @@
 
 this_file = Path(__file__)
```
Comment (JCZuurmond) [inline tests/integration/source_code/solacc.py:24]: I expect globals to be in ALL_CAPS

**Review by JCZuurmond — APPROVED** [review]
Some nits

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**